crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -1691,43 +1691,34 @@ impl<'db> TypeInferenceBuilder<'db> {
         self.infer_annotation_expression(&type_alias.value, DeferredExpressionState::Deferred);
     }
 
-    /// Returns `true` if the current scope is the function body scope of a method of a protocol
-    /// (that is, a class which directly inherits `typing.Protocol`.)
-    fn in_protocol_class(&self) -> bool {
```

Comment:
Given that there were two different uses of this method, and both now become significantly more verbose, would it really be a problem to keep this method as short-hand, even though it's body would now be a one-liner?

Reply:
In one of the two use cases (the one where we now want to retain the enclosing class context even if the class is not a protocol class), I think it would be a mistake to use this method, because you'd end up doing a lot of work twice in quick succession. So I feel like it's sort-of a footgun to keep it?